### DISCUSSION TITLE
Information on specific android payload

### DISCUSSION THREAD

**Discussion by Morsmalleo** [rapid7/metasploit-framework#16647] [Q&A] [answered]
G'day guys, just another quick question here. 

I was wondering, when we rig up the *msfvenom* to use the android payload `android/meterpreter/reverse_tcp`, what are the exact paths to the modules, payloads, stagers etc... that the *msfconsole* uses in order to build a full functioning metasploit reverse_tcp APK from this option??

If anyone can answer this it would be greatly appreciated

**Comment by bcoles** [comment, marked as answer]
The Metasploit library for constructing APK files is here:

* https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/payload/apk.rb

The android payload libraries are here:

* https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/android.rb
* https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/payload/android.rb

The android payload wrappers are here:

* https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/payload/android/
* https://github.com/rapid7/metasploit-framework/tree/master/modules/payloads

The android payloads utilise payloads from the `metasploit-payloads` repository:

* https://github.com/rapid7/metasploit-payloads/tree/master/java/androidpayload
* https://github.com/rapid7/metasploit-payloads/tree/master/java/meterpreter/meterpreter/src/main/java/com/metasploit/meterpreter

**Reply by Morsmalleo** [reply]
Thank you very much

**Reply by Morsmalleo** [reply]
That definitely points me in the right direction
